picasso/src/main/java/com/squareup/picasso/Request.java

```diff
@@ -500,6 +497,10 @@ public Request build() {
         throw new IllegalStateException(
             "Center inside requires calling resize with positive width and height.");
       }
+      if (onlyScaleDown && (targetWidth == 0 && targetHeight == 0)) {
```

Comment:
Ahh, the `Request` class has no knowledge of fitting, which is done in the `RequestCreator` class. I'm going to make my own pull request that will hopefully fix these issues cleanly.